setupext.py

```diff
@@ -691,15 +692,31 @@ def do_custom_build(self, env):
                 f.write(vcxproj)
 
             cc = get_ccompiler()
-            cc.initialize()  # Get msbuild in the %PATH% of cc.spawn.
+            cc.initialize()
+            # On setuptools versions that use "local" distutils,
+            # ``cc.spawn(["msbuild", ...])`` no longer manages to locate the
+            # right executable, even though they are correctly on the PATH,
+            # because only the env kwarg to Popen() is updated, and not
+            # os.environ["PATH"]. Instead, use shutil.which to walk the PATH
+            # and get absolute executable paths.
+            with TemporaryDirectory() as tmpdir:
+                dest = Path(tmpdir, "path")
+                cc.spawn([
```

Comment:
Why the extra subprocess call?

Reply:
See above comment; this is copied from mplcairo.